packages/service-library/src/servicelib/utils.py

```diff
@@ -141,3 +146,24 @@ def ensure_ends_with(input_string: str, char: str) -> str:
     if not input_string.endswith(char):
         input_string += char
     return input_string
+
+
+def slice_list_iter(
+    input_list: list[T], *, slice_size: NonNegativeInt
```

Comment:
MINOR: why you constraint lists to single type or items `T` ?  This algorithm can also work with a hybrid-type list like `["a", 1,2, True]`.  I would just annotate it as `list` (i.e. drop the `[T]`).

Reply:
ok, we can drop the typing.